src/main.c

```diff
@@ -271,30 +344,66 @@ start_in_save_mode(GHashTable *ips_to_migrate, bool *stop_flag)
 }
 
 /**
- * Creates hashtable of IP addresses from CLI arguments.
+ * Start threads required for save mode of operation.
+ * The workflow is used to maintain a local copy of the CT entries for
+ * the VM.
+ *
+ * Args:
+ *   @ips_to_migrate Hashtable of IP addresses for which CT entries
+ *                   have to be migrated.
+ *   @ct_zones_to_migrate Hashtable of Ct zones for which CT entries have to be
+ *                        migrated.
+ *   @stop_flag flag used by thread to exit after dbus operations.
+ *
+ * Returns:
+ *   0 in case of success, -1 otherwise
+ */
+static int
+start_in_save_mode(GHashTable *ips_to_migrate, GHashTable *ct_zones_to_migrate,
+                   bool *stop_flag)
+{
+    if ( fltr_type == FILTER_BY_IP ) {
+        return start_in_save_mode_for_ip(ips_to_migrate, stop_flag);
+    }
+
+    return start_in_save_mode_for_ct_zones(ct_zones_to_migrate, stop_flag);
+}
+
+
+/**
+ * Creates hashtable of IP addresses/ CT Zones from CLI arguments.
  *
  * Args:
  *   @argv array of CLI args.
  *
  * Returns:
- *   Resulting hashtable containing IP address(uint32_t) as key.
+ *   Resulting hashtable containing IP address/CT zones(uint32_t)  as key.
  */
-static GHashTable *
-create_ips_ht_from_args(char *argv[])
+GHashTable *
+get_entity_ht_from_args(char *argv[])
```

Comment:
since we are using if else to determine whether to form ips or ct-zone,

better to keep the original function as it is.
and define a new function `create_ct_zones_ht_from_args`

I don't see any code being reused here.